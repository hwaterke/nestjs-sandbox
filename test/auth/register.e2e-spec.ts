import {Test, TestingModule} from '@nestjs/testing'
import {gql} from 'apollo-server-express'
import {Connection} from 'typeorm'
import {AppModule} from '../../src/app.module'
import {TestingUtils} from '../utils/testing.utils'

const REGISTER = gql`
  mutation register($email: String!, $password: String!) {
    register(email: $email, password: $password) {
      user {
        email
      }
      token
    }
  }
`

describe('AuthResolver.register (e2e)', () => {
  let connection: Connection
  let testUtils: TestingUtils

  beforeAll(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile()

    testUtils = await TestingUtils.initNestApplication(moduleFixture)

    connection = moduleFixture.get(Connection)
  })

  afterEach(() => {
    testUtils.token = null
    return connection.synchronize(true)
  })

  afterAll(() => {
    return testUtils.close()
  })

  it('should be allowed to register', async () => {
    const result = await testUtils.apolloClient.mutate({
      mutation: REGISTER,
      variables: {
        email: 'test@example.com',
        password: '12345678',
      },
    })

    expect(result.errors).toBeUndefined()
    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
    expect(result.data.register.user).toMatchSnapshot()
    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
    expect(result.data.register.token).toBeDefined()
  })

  it('should not be allowed to register with an invalid email', async () => {
    const result = await testUtils.apolloClient.mutate({
      mutation: REGISTER,
      variables: {
        email: 'test',
        password: '12345678',
      },
    })

    expect(result).toMatchSnapshot()
  })

  it('should not be allowed to register with password too short', async () => {
    const result = await testUtils.apolloClient.mutate({
      mutation: REGISTER,
      variables: {
        email: 'test@example.com',
        password: '1234567',
      },
    })

    expect(result).toMatchSnapshot()
  })

  it('should not be allowed to register with an already used email', async () => {
    await testUtils.apolloClient.mutate({
      mutation: REGISTER,
      variables: {
        email: 'test@example.com',
        password: '12345678',
      },
    })

    const result = await testUtils.apolloClient.mutate({
      mutation: REGISTER,
      variables: {
        email: 'test@example.com',
        password: '12345678',
      },
    })

    expect(result).toMatchSnapshot()
  })
})
