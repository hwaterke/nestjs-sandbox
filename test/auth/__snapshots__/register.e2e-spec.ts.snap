// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthResolver.register (e2e) should be allowed to register 1`] = `
Object {
  "email": "test@example.com",
}
`;

exports[`AuthResolver.register (e2e) should not be allowed to register with an already used email 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: EMAIL_ALREADY_USED],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`AuthResolver.register (e2e) should not be allowed to register with an invalid email 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Bad Request Exception],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`AuthResolver.register (e2e) should not be allowed to register with password too short 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Bad Request Exception],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
